tutorials/users/ide/kernels-and-dependency-management.md

```diff
@@ -1 +1,108 @@
+environments available to you.  
+
+Below are examples of kernels available on Marble.  These will be updated or changed to fit the needs of the users.
+
+![Kernels](images/kernels-dependency/kernel-buttons.png)
+
+## How to install packages
+
+The kernels on Marble are created with a focus on climate science and data analysis.  As such, they are installed with 
+programming and scripting languages commonly used for this purpose, such as Python and R.
+
+For the purpose of this tutorial, Python will be used in the examples.
+
+Start off by clicking on the Console button for the kernel you want to install a package for.
+
+![Console Buttons](images/kernels-dependency/console-buttons.png)
+
+This will open a Console tab where you will be able to enter commands at the bottom.  Execute the commands by using 
+"Shift+Enter" on your keyboard.
+
+### List the packages already installed
+
+First, find out which packages are already installed. There are several ways to see the packages already installed in a 
+kernel: `pip list`, `pip freeze`, `conda list`
+
+#### `pip list`
+
+`pip list` lists all the packages installed and its version.  It will also show the packages used for package management, such as `setuptools`.
+
+![pip list](images/kernels-dependency/pip-list-list.png)
+
+#### `pip freeze`
+
+`pip freeze` will list all the packages installed with the `pip` command and its version.  Sometimes it will also
+show the VCS url if the package source is from a particular version control system.
+
+The benefit of `pip freeze` is it outputs in the same format used by the `requirements.txt` file, a 
+configuration file used for bulk package installation.  If you need to create a requirements file, use 
+`pip freeze` with `>` to direct the output to a file.
+
+```
+pip freeze > /path/to/requirements.txt
+```
+
+![pip freeze](images/kernels-dependency/pip-freeze-list.png)
+
+```{note}
+If `pip freeze` shows the VCS url instead of the version of the package, use `pip list` with the freeze format
+```
+
+```
+pip list --format=freeze > requirements.txt
+```
+
+#### `conda list`
+
+`conda list` will list all packages installed with the `conda` command and it's version, along with the build channel 
+the package was installed from.  It will also show the kernel (environment) the packages are installed under. 
+
+```{note}
+You will only be able to use the `conda` command if you have [Anaconda](https://www.anaconda.com/) or one of its derivatives installed.
+```
+
+![conda](images/kernels-dependency/conda-list-list.png)
+
+
+### Install packages individually
+
+To install a package you need the package's name.  An excellent resource is the [Python Package Index](https://pypi.org/) (pypi).
+
+The results of the *pypi* search includes the syntax of the install command.
+
+![pypi](images/kernels-dependency/pypi-result.png)
```

Comment:
Needs some additional explanation (users don't know what alembic is). 

> For example, searching for the `alembic` package on pypi.org shows the following version information and installation command for that package:

Reply:
Added explanation for Alembic